airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -47,6 +47,10 @@ def __init__(self, shop: str, api_key: str, api_password: str, api_version: str,
         self.api_version = api_version
         self.since_id = 0
 
+    @property
+    def url_base(self) -> str:
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/"
```

Comment:
Manage to achieve this only by changing the @property of the HttpStream class, like above, since when I use simple hardcode a.k.a url_base = "..." it throws the error "value is not-accessible". Is it bad to use it by changing the property of the class like os?